kustomize/commands/edit/add/addcomponent_test.go

```diff
@@ -52,7 +52,7 @@ func TestAddComponentAlreadyThere(t *testing.T) {
 }
 
 func TestAddKustomizationFileAsComponent(t *testing.T) {
```

Comment:
I think this test was originally supposed to check that if you try to add the kustomization.yaml file itself to `components`, it won't get added.

@annasong20 attempted to fix this in https://github.com/kubernetes-sigs/kustomize/pull/4702/files#diff-6f49ffbefe3aa6f9f90b63fa94b7019f1004257536c2264e6445f7b311d10b99, but it fell off our radar and it never merged. Do you mind picking up her fixes from `TestAddKustomizationFileAsComponent` and `TestAddKustomizationFileAsResource` from that PR?

Reply:
I added it at https://github.com/kubernetes-sigs/kustomize/pull/5030/commits/565cff2d07dad94b66729ddb9260c13b53442528.